src/r_tipping/RateSystem.jl

```diff
@@ -0,0 +1,133 @@
+- `forcing_scale = 1.0`: Amplitude of the ramping. The ramping is then automatically rescaled 
+- `t0 = 0.0`: Initial time of the resulting non-autonomous system (relevant to later compute trajectories)
+
+## Description
+The returned `RateSystem.system` is is 
+autonomous before `forcing_start`, 
+non-autnonmous from `forcing_start` to `forcing_start+forcing_length` with the parameter shift given by the [`RateConfig`](@def), and again 
+autonomous after `forcing_start+forcing_length`.
+
+## Fields of the resulting `RateSystem` type
+- `system`: Nonautonomous [`CoupledODEs`](@ref) constructed from an autonomous system `sys` and a `RateConfig`
+- `forcing`: Function giving the value of the ramped parameter for every time.
+- `pidx`: Index of the ramped parameter within the parameter container of the autonomous system `sys`.
+
+"""
+function RateSystem(
+    auto_sys::ContinuousTimeDynamicalSystem,
+    rc::RateConfig,
+    pidx;
+    forcing_start=rc.section_start,
+    forcing_length=(rc.section_end - rc.section_start),
+    forcing_scale=1.0,
+    t0=0.0,
+)
+    params = deepcopy(current_parameters(auto_sys))
+    p0 = params[pidx]
+
+    if forcing_start < t0
+        @warn "The forcing starts before the system initial time t0=$(t0)."
+    end
+
+    system = apply_ramping(
+        auto_sys, rc, pidx, p0, params, forcing_start, forcing_length, forcing_scale, t0
+    )
+    forcing = t -> p_modified(t, rc, p0, forcing_start, forcing_length, forcing_scale)
+
+    return RateSystem(system, forcing, pidx)
+end
+
+## Creates a piecewise constant function in alignment with the entries of the RateConfig and the parameter value of the underlying autonomous system
+function p_modified(
+    t::Real, rc::RateConfig, p0, forcing_start, forcing_length, forcing_scale
+)
+    p = rc.pfunc
+    section_start = rc.section_start
+    section_end = rc.section_end
+
+    # making the function piecewise constant with range [p0,p0+forcing_scale]    
+    q = if t ≤ forcing_start
+        return p0
+    else
+        if t < forcing_start+forcing_length
+            # performing the time shift corresponding to stretching/squeezing 
+            time_shift =
+                ((section_end-section_start)/forcing_length)*(t-forcing_start)+section_start
+            return p0 +
+                   forcing_scale*(
+                p(time_shift)-p(section_start)
+            )/(p(section_end)-p(section_start))
+        else
+            return p0 + forcing_scale
+        end
+    end
+
+    return q
+end
+
+# returns a continuous time dynamical system with modified drift field
+function apply_ramping(
+    auto_sys, rc, pidx, p0, params, forcing_start, forcing_length, forcing_scale, t0
+)
+    f = deepcopy(dynamic_rule(auto_sys))
+
+    function f_new(u, p, t)
```

Comment:
In the email you said that this is 3x slower...? Wow. This is very bad! We should brainstorm a bit alternative ways to do this. Typically you'd hope to have a slowdown of 10-50% when going from a hard-coded to a general solution but here we are talking about a slowdown of 300% which is massive. @reykboerner can you also help tjhe brainstorming please? I'll also think of something.